src/Core.UnitTests/Binding/NonRoslynBindingConfigProviderTests.cs

```diff
@@ -289,8 +289,8 @@ public void IsSupported()
         }
 
         private static SonarQubeRule CreateRule(string ruleKey, string repoKey, bool isActive,
-            SonarQubeIssueSeverity severity, IDictionary<string, string> parameters = null, IReadOnlyList<SonarQubeDescriptionSection> descriptionSections = null, IReadOnlyList<string> educationPrinciples = null) =>
-            new SonarQubeRule(ruleKey, repoKey, isActive, severity, parameters, SonarQubeIssueType.Unknown, descriptionSections, educationPrinciples);
+            SonarQubeIssueSeverity severity, IDictionary<string, string> parameters = null, string description = null, IReadOnlyList<SonarQubeDescriptionSection> descriptionSections = null, IReadOnlyList<string> educationPrinciples = null) =>
```

Comment:
It looks like you have two unrelated changes in this PR - adding the `description` field, and adding the new API call. More than half of the changes in the PR relate to the `description` field, which has nothing to do with subject of the PR.

In future it would be better to make the changes in separate PRs. This improves traceability as well as making the PRs simpler.